app/questionnaire/value_source_resolver.py

```diff
@@ -116,6 +121,33 @@ def _resolve_answer_value_source(
 
         return answer_value
 
+    def _resolve_progress_value_source(
+        self, value_source: Mapping
+    ) -> ValueSourceEscapedTypes | ValueSourceTypes | None:
+        identifier = value_source["identifier"]
+        selector = value_source["selector"]
+        if selector == "section":
+            return self.progress_store.get_section_status(
+                section_id=identifier, list_item_id=None
```

Comment:
Worth adding a comment to say we don't support list item is none because we don't support checking the progress of a repeating section.